Lib/sysconfig.py

```diff
@@ -204,20 +224,13 @@ def joinuser(*args):
     return joinuser("~", ".local")
 
 
-def _parse_makefile(filename, vars=None):
+def _parse_makefile(filename, vars=None, distutils_compat=False):
     """Parse a Makefile-style file.
 
     A dictionary containing name/value pairs is returned.  If an
     optional dictionary is passed in as the second argument, it is
     used instead of a new dictionary.
     """
-    # Regexes needed for parsing Makefile (and similar syntaxes,
-    # like old-style Setup files).
-    import re
```

Comment:
It’s possible that `re` was imported in this function rather than at module scope because building CPython (and the _sre modules) needs sysconfig.

Reply:
Ok, I'll add try/except to handle situation when re module won't be available. Or do you have a better idea?

Reply:
That may not work if the problem is circular imports. Better to just not use re until it's needed (and there's probably nothing gained from `re.compile` here anyway; the regular re methods will cache well enough).

Reply:
That would mean import re and define some of the regexes in three places:
* expand_makefile_vars function in sysconfig
* _parse_makefile function in sysconfig
* globally in distutils.sysconfig because they're imported from there in read_setup_file function in distutils.extension (not future proof)

I'd rather keep them in one place. I've checked and there seems to be no circular import. Do you think there might be a problem which the fresh builds in CI cannot spot?